monitoring/ceph-mixin/alerts.libsonnet

```
@@ -0,0 +1,875 @@
+            'for': '30s',
+            labels: {
+              severity: 'critical',
+              type: 'ceph_default',
+              oid: '1.3.6.1.4.1.50495.1.2.1.3.1',
+            },
+            annotations: {
+              documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#mon-down',
+              summary: 'Monitor quorum is at risk',
+              description: '{{ $min := query "floor(count(ceph_mon_metadata) / 2) +1" | first | value }}Quorum requires a majority of monitors (x {{ $min }}) to be active\nWithout quorum the cluster will become inoperable, affecting all connected clients and services.\n\nThe following monitors are down:\n{{- range query "(ceph_mon_quorum_status == 0) + on(ceph_daemon) group_left(hostname) (ceph_mon_metadata * 0)" }}\n  - {{ .Labels.ceph_daemon }} on {{ .Labels.hostname }}\n{{- end }}\n',
+            },
+          },
+          {
+            alert: 'CephMonDown',
+            expr: '(count(ceph_mon_quorum_status == 0) <= (count(ceph_mon_metadata) - floor(count(ceph_mon_metadata) / 2) + 1))',
+            'for': '30s',
+            labels: {
+              severity: 'warning',
+              type: 'ceph_default',
+            },
+            annotations: {
+              documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#mon-down',
+              summary: 'One of more ceph monitors are down',
+              description: '{{ $down := query "count(ceph_mon_quorum_status == 0)" | first | value }}{{ $s := "" }}{{ if gt $down 1.0 }}{{ $s = "s" }}{{ end }}You have {{ $down }} monitor{{ $s }} down.\nQuorum is still intact, but the loss of further monitors will make your cluster inoperable.\n\nThe following monitors are down:\n{{- range query "(ceph_mon_quorum_status == 0) + on(ceph_daemon) group_left(hostname) (ceph_mon_metadata * 0)" }}\n  - {{ .Labels.ceph_daemon }} on {{ .Labels.hostname }}\n{{- end }}\n',
+            },
+          },
+          {
+            alert: 'CephMonDiskspaceCritical',
+            expr: 'ceph_health_detail{name="MON_DISK_CRIT"} == 1',
+            'for': '1m',
+            labels: {
+              severity: 'critical',
+              type: 'ceph_default',
+              oid: '1.3.6.1.4.1.50495.1.2.1.3.2',
+            },
+            annotations: {
+              documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#mon-disk-crit',
+              summary: 'Disk space on at least one monitor is critically low',
+              description: "The free space available to a monitor's store is critically low (<5% by default).\nYou should increase the space available to the monitor(s). The\ndefault location for the store sits under /var/lib/ceph. Your monitor hosts are;\n{{- range query \"ceph_mon_metadata\"}}\n  - {{ .Labels.hostname }}\n{{- end }}\n",
```

One thing I love from jsonnet is verbatims (but YAML supports that too):
```suggestion
              description: |||
The free space available to a monitor's store is critically low (<5% by default).
You should increase the space available to the monitor(s). The
default location for the store sits under /var/lib/ceph. Your monitor hosts are;
{{- range query "ceph_mon_metadata"}}
  - {{ .Labels.hostname }}
{{- end }}
|||,
```

---

I was also planning to do this kind of cleanup in a followup PR (same as https://github.com/ceph/ceph/pull/44059#discussion_r770604922)

---

Perfect! Yeah, given the size of this PR no need to make it heavier.